plugin-store/src/components/plugins/Detail.tsx

```diff
@@ -0,0 +1,254 @@
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+      'Forward-To': url,
+    },
+  })
+    .then(response => {
+      if (response.ok) {
+        return response.json();
+      }
+    })
+    .catch(error => {
+      throw new Error(`Request failed!${error}`);
+    });
+}
+
+async function checkIfPluginIsInstalled(pluginDetail: any) {
+  try {
+    const headlampPlugin = runCommand('electron-node', ['plugin-management.js', 'list', '-j']);
+    const data = await new Promise<string>((resolve, reject) => {
+      headlampPlugin.stdout.on('data', (data: string) => {
+        resolve(data);
+      });
+      headlampPlugin.stderr.on('data', (data: string) => {
+        reject(data);
+      });
+    });
+
+    const parsedData = JSON.parse(data) as any[];
+
+    const currentPlugin =
+      parsedData.find(plugin => plugin.folderName === pluginDetail.name) || null;
+    pluginDetail.isInstalled = false;
+    pluginDetail.currentVersion = '';
+    pluginDetail.packageName = '';
+    pluginDetail.updateAvailable = false;
+
+    if (currentPlugin) {
+      pluginDetail.isInstalled = true;
+      pluginDetail.packageName = currentPlugin?.pluginName;
+      pluginDetail.currentVersion = currentPlugin?.artifacthubVersion;
+      if (semver.gt(pluginDetail.version, currentPlugin?.artifacthubVersion)) {
+        pluginDetail.updateAvailable = true;
+      }
+    }
+
+    return pluginDetail;
+  } catch (error) {
+    return pluginDetail;
+  }
+}
+
+export function PluginDetail() {
+  const { repoName, pluginName } = useParams<{ repoName: string; pluginName: string }>();
+  const [pluginDetail, setPluginDetail] = useState<any>(null);
+  const [currentAction, setCurrentAction] = useState<'Install' | 'Uninstall' | 'Update' | null>(
+    null
+  );
+  const [currentActionMessage, setCurrentActionMessage] = useState<string | null>(null);
+  const [currentActionState, setCurrentActionState] = useState<string | null>(null);
+  const [openDialog, setOpenDialog] = useState(false);
+
+  const url = `https://artifacthub.io/packages/headlamp/${repoName}/${pluginName}`;
+
+  function handleInstall(url) {
+    setOpenDialog(true);
+    setCurrentAction('Install');
+    const headlampPlugin = runCommand('electron-node', [
+      'plugin-management.js',
+      'install',
+      url,
+      '-j',
+    ]);
+
+    headlampPlugin.stdout.on('data', (data: string) => {
+      const parsedData = JSON.parse(data) as any;
+      setCurrentActionMessage(parsedData.message);
+      setCurrentActionState(parsedData.status);
+    });
+    headlampPlugin.stderr.on('data', (data: string) => {
+      setCurrentActionMessage(data);
+      setCurrentActionState('error');
+    });
+  }
+
+  function handleUpdate(pluginName) {
+    setOpenDialog(true);
+    setCurrentAction('Uninstall');
+
+    const headlampPlugin = runCommand('electron-node', [
+      'plugin-management.js',
+      'update',
+      pluginName,
+      '-j',
+    ]);
```

Comment:
I think this should be rather a new `Headlamp.` API call that does this transparently for the plugin.